src/NLog/SetupBuilderExtensions.cs

```diff
@@ -107,6 +115,16 @@ public static ISetupBuilder LoadConfiguration(this ISetupBuilder setupBuilder, A
                 setupBuilder.LogFactory.Configuration = newConfig;
             }
 
+            if (applyOnReload)
+            {
+#if !NETSTANDARD1_3
+                setupBuilder.LogFactory.ConfigurationChanged += (sender, e) =>
```

Comment:
I'm always a bit confused by Changed and Reloaded :'(

Seems Reloaded only works for file reloads?

Not sure if I need a test for that case (file reload) then

Reply:
How do you ensure that you don't have multiple callbacks? Think users will expect that previous delegates will be reset, if calling LoadConfiguration twice.

Reply:
Could you give an example?
```
  logFactory.Setup().
                LoadConfigurationFromFile().
                LoadConfiguration(true, b =>
                {
                    b.Configuration.AddRuleForAllLevels("target1");
                })
                .LoadConfiguration(true, b =>
                {
                    b.Configuration.AddRuleForAllLevels("target2");
                });
```

Reply:
I was just thinking it might confuse if doing Setup()-operation more than once. But lets see what happens. Btw. can you make sure that the callback is only called when a new config is loaded? (Avoid callback with null on shutdown / configuration assigned to null).

Reply:
> might confuse if doing Setup()-operation more than once

No this is great! Always wanted this! OK, maybe it needs a reset, but you could do that in LoadConfiguration